### DISCUSSION TITLE
Self-Hosting: Unable to Create Account - AxiosError: Network Error

### DISCUSSION THREAD

**Discussion by kojid0** [ente-io/ente#1449] [Q&A]
I was able to set up a self-hosted instance of ente following the [instructions](https://help.ente.io/self-hosting/) (running on a debian12 server).
Unfortunately, I get an Unhandled Runtime Error (AxiosError: Network Error) when I try to create an account. 
![grafik](https://github.com/ente-io/ente/assets/24477557/c97330e3-2424-43d2-bd32-63fe425ed4a5)

The server has internet access, otherwise the installation of ente would have failed. But I haven't set up a SMTP Email-Server yet. Maybe this is the reason? How to do this?

**Comment by mnvr** [comment]
> But I haven't set up a SMTP Email-Server yet. Maybe this is the reason? 

No, the SMTP server is not the reason, that only comes later when you need an OTP. You also don't need to set an SMTP server if you just need an OTP code, there are other ways as mentioned here - https://help.ente.io/self-hosting/faq/otp

--

> The server has internet access, otherwise the installation of ente would have failed. 

The installation worked, but you still need to ask the web app that you're running to connect to the right endpoint. I think the issue here is that the web app is not connecting to the correct endpoint.

e.g. if you run the web app this way
```sh
NEXT_PUBLIC_ENTE_ENDPOINT=http://localhost:8080 yarn dev
```

then you'll need to make sure that your server is indeed running on `http://localhost:8080` (or whatever is the IP of the instance your running it on).

You don't even need to run the web app to verify connectivity. e.g. you can run

```sh
curl http://localhost:8080/ping
```

to check that the `NEXT_PUBLIC_ENTE_ENDPOINT` you are setting is reachable from where you are trying to run the web app.

**Reply by acmartinsalmeida** [reply]
Hi !
Same error here

Even if 

From server

curl http://localhost:8080/ping
{"id":"development-cluster","message":"pong"}

The only difference is that i'm using IP of server instead of localhost to reach webapp eg : http://192.168.1.xxx:3000

Maybe a problem with cors ?

I've set up nginx with static web app and i get the same error

SMTP is indeed configured

Does @kojid0 succed ?

**Reply by mnvr** [reply]
You will also need to provide the IP of the the server (museum) to the webapp when you're starting the web app

e.g.
```
NEXT_PUBLIC_ENTE_ENDPOINT=http://192.168.1.xxx:8080 yarn dev
```

`NEXT_PUBLIC_ENTE_ENDPOINT` tells the web app where it should connect to for making API requests. Since you're getting axios errors, it means the web app is not able to reach the correct place. SMTP is not the issue.

**Reply by acmartinsalmeida** [reply]
Thank you for point me out to this. 
I've setup my 3 s3 local and now i can upload..
But it does not show thumbnail

![Capture d’écran du 2024-04-28 21-36-54](https://github.com/ente-io/ente/assets/1231877/7661fb91-5a57-48b6-b866-3baf27ec565c)

What i miss ?

**Reply by mnvr** [reply]
That is a bit surprising, I'm not sure why your thumbnail cache is not getting created. I've modified the code so that it works even without the thumbnail cache - https://github.com/ente-io/ente/pull/1552 - so if you pull the latest main branch, that issue would go away.

But it would be great if you could try and figure out why your thumbnailCache is not being initialized (this issue is not related to self hosting, maybe it is something specific to your browser). Could you tell the following information:
* What browser are you using
* If you are using private mode

In the logs (Sidebar > Debug logs), could you also see if you get some error message when the app tries to initialize this thumbnail cache. You can look for a line like this "Failed to open thumbnail cache, will continue without it" which will be followed by the error. Thank you!

**Reply by acmartinsalmeida** [reply]
It works now ^^

My browser is Firefox 125.0.2 (64 bits) on Debian Bookworm

I'm not using private mode

In Sidebar > Debug logs

[2024-04-29T05:21:47.197Z] [error] Failed to open thumbnail cache, will continue without it: ReferenceError: caches is not defined
openWebCache@webpack-internal:///../../packages/next/blob-cache.ts:65:19
openCache@webpack-internal:///../../packages/next/blob-cache.ts:22:117
init@webpack-internal:///./src/services/download/index.ts:39:100
main@webpack-internal:///./src/pages/gallery/index.tsx:261:78

Ente is not installed at localhost. It is installed on a server (Debian Bookworm too) on the same network  

hope it helps

**Reply by cristiandavidgm** [reply]
"caches is not defined" <- if you are using chrome 'caches' is only available if using HTTPS.
